### PR TITLE
a 2nd set of cleanups and optimizations

### PR DESCRIPTION
**Pull request by jfmherokiller**

### CODE REVIEW

File: Sources/Plasma/Apps/plClient/win32/plClient_Win.cpp [hunk 1]
```diff
@@ -99,14 +99,12 @@ void plClient::IChangeResolution(int width, int height)
     HMONITOR monitor = MonitorFromWindow(fWindowHndl, MONITOR_DEFAULTTONULL);
     if (!monitor)
         return;
-    MONITORINFOEXW moninfo;
-    memset(&moninfo, 0, sizeof(moninfo));
+    MONITORINFOEXW moninfo = {};
```
Comment (Hoikas) [inline Sources/Plasma/Apps/plClient/win32/plClient_Win.cpp:102]:
```suggestion
    MONITORINFOEXW moninfo{};
```

File: Sources/Plasma/Apps/plClient/win32/plClient_Win.cpp [hunk 2]
```diff
@@ -99,14 +99,12 @@ void plClient::IChangeResolution(int width, int height)
...
     moninfo.cbSize = sizeof(moninfo);
     GetMonitorInfoW(monitor, &moninfo);
 
     // Fetch a base display settings
-    DEVMODEW devmode;
-    memset(&devmode, 0, sizeof(devmode));
+    DEVMODEW devmode = {};
```
Comment (Hoikas) [inline Sources/Plasma/Apps/plClient/win32/plClient_Win.cpp:107]:
```suggestion
    DEVMODEW devmode{};
```

File: Sources/Plasma/FeatureLib/pfConsole/pfConsoleDirSrc.cpp [hunk 3]
```diff
@@ -61,13 +61,13 @@ bool pfConsoleDirSrc::ParseDirectory(const plFileName& path, const char* mask /*
     hsAssert(fEngine != nullptr, "Cannot do a dir execute without an engine!");
 
     std::vector<plFileName> files = plFileSystem::ListDir(path, mask);
-    for (auto iter = files.begin(); iter != files.end(); ++iter)
+    for (auto& file : files)
```
Comment (Hoikas) [inline Sources/Plasma/FeatureLib/pfConsole/pfConsoleDirSrc.cpp:64]:
```suggestion
    for (const auto& file : files)
```

File: Sources/Plasma/FeatureLib/pfConsole/pfConsoleDirSrc.cpp [hunk 4]
```diff
@@ -61,13 +61,13 @@ bool pfConsoleDirSrc::ParseDirectory(const plFileName& path, const char* mask /*
...
     {
-        plFileName name = iter->GetFileName();
+        plFileName name = file.GetFileName();
```
Comment (Hoikas) [inline Sources/Plasma/FeatureLib/pfConsole/pfConsoleDirSrc.cpp:66]: `file` is already a `plFileName`. `GetFileName()` returns an `ST::string`, so we are making an unnecessary copy here.

File: Sources/Plasma/PubUtilLib/plAgeDescription/plAgeDescription.cpp [hunk 5]
```diff
@@ -491,11 +491,8 @@ void    plAgeDescription::CopyFrom(const plAgeDescription& other)
...
 {
-    for (const plAgePage& page : fPages)
+    return std::any_of(fPages.begin(), fPages.end(), [this, loc](const plAgePage& page)
     {
-        plLocation pageLoc = CalcPageLocation(page.GetName());
-        if (pageLoc == loc)
-            return true;
-    }
-    return false;
+	    return loc == CalcPageLocation(page.GetName());
```
Comment (Hoikas) [inline Sources/Plasma/PubUtilLib/plAgeDescription/plAgeDescription.cpp:496]:
```suggestion
	    return CalcPageLocation(page.GetName()) == loc;
```

Let's not invert the order of the comparison.

File: Sources/Plasma/PubUtilLib/plAgeDescription/plAgeDescription.cpp [hunk 6]
```diff
@@ -491,11 +491,8 @@ void    plAgeDescription::CopyFrom(const plAgeDescription& other)
 
 bool plAgeDescription::FindLocation(const plLocation& loc) const
 {
-    for (const plAgePage& page : fPages)
+    return std::any_of(fPages.begin(), fPages.end(), [this, loc](const plAgePage& page)
```
Comment (Hoikas) [inline Sources/Plasma/PubUtilLib/plAgeDescription/plAgeDescription.cpp:494]:
```suggestion
    return std::any_of(fPages.begin(), fPages.end(), [this, &loc](const plAgePage& page)
```

File: Sources/Plasma/PubUtilLib/plAnimation/plAGAnim.cpp [hunk 7]
```diff
@@ -438,17 +438,17 @@ void plATCAnim::Write(hsStream *stream, hsResMgr *mgr)
     stream->WriteLEFloat(fEaseOutLength);
 
     stream->WriteLE32((uint32_t)fMarkers.size());
-    for (MarkerMap::iterator it = fMarkers.begin(); it != fMarkers.end(); it++)
+    for (auto& fMarker : fMarkers)
```
Comment (Hoikas) [inline Sources/Plasma/PubUtilLib/plAnimation/plAGAnim.cpp:441]:
```suggestion
    for (const auto& fMarker : fMarkers)
```

File: Sources/Plasma/PubUtilLib/plAnimation/plAGAnim.cpp [hunk 8]
```diff
@@ -438,17 +438,17 @@ void plATCAnim::Write(hsStream *stream, hsResMgr *mgr)
...
     {
-        stream->WriteSafeString(it->first);
-        stream->WriteLEFloat(it->second);
+        stream->WriteSafeString(fMarker.first);
+        stream->WriteLEFloat(fMarker.second);
     }
 
     stream->WriteLE32((uint32_t)fLoops.size());
-    for (LoopMap::iterator loopIt = fLoops.begin(); loopIt != fLoops.end(); loopIt++)
+    for (auto& fLoop : fLoops)
```
Comment (Hoikas) [inline Sources/Plasma/PubUtilLib/plAnimation/plAGAnim.cpp:448]:
```suggestion
    for (const auto& loop : fLoops)
```

The `f` prefix is reserved for class `f`ields.

File: Sources/Plasma/PubUtilLib/plAnimation/plAGAnim.cpp [hunk 9]
```diff
@@ -438,17 +438,17 @@ void plATCAnim::Write(hsStream *stream, hsResMgr *mgr)
...
     {
-        stream->WriteSafeString(loopIt->first);
-        std::pair<float,float>& loop = loopIt->second;
+        stream->WriteSafeString(fLoop.first);
+        std::pair<float,float>& loop = fLoop.second;
```
Comment (Hoikas) [inline Sources/Plasma/PubUtilLib/plAnimation/plAGAnim.cpp:451]:
```suggestion
        stream->WriteSafeString(loop.first);
        std::pair<float,float>& loop = loop.second;
```

File: Sources/Plasma/PubUtilLib/plAvatar/plAvBrainHuman.cpp [hunk 10]
```diff
@@ -304,20 +299,14 @@ void plAvBrainHuman::IInitBoneMap()
...
-
-    for(int i = 0; i < numTuples; i++)
+    for (const auto& [BoneID, BoneName] : BoneMap)
     {
-        HumanBoneID id = tupleMap[i].fID;
-        ST::string name = tupleMap[i].fName;
-        
-        const plSceneObject * bone = this->fAvMod->FindBone(name);
-        if( bone )
+	    if( const plSceneObject * bone = this->fAvMod->FindBone(BoneName) )
```
Comment (Hoikas) [inline Sources/Plasma/PubUtilLib/plAvatar/plAvBrainHuman.cpp:304]:
```suggestion
	    if (const plSceneObject * bone = fAvMod->FindBone(BoneName))
```

File: Sources/Plasma/PubUtilLib/plAvatar/plAvBrainHuman.cpp [hunk 11]
```diff
@@ -304,20 +299,14 @@ void plAvBrainHuman::IInitBoneMap()
...
         {
-            fAvMod->AddBoneMapping(id, bone);
+            fAvMod->AddBoneMapping(BoneID, bone);
         }
         else
-            hsStatusMessageF("Couldn't find standard bone %s.", name.c_str());
+            hsStatusMessageF("Couldn't find standard bone %s.", BoneName.data());
```
Comment (Hoikas) [inline Sources/Plasma/PubUtilLib/plAvatar/plAvBrainHuman.cpp:309]: While we're here, let s fix this bracing to match [Plasma Code Style](https://github.com/H-uru/Plasma/wiki/Preferred-Code-Style). We prefer for all branches to use brackets if any one branch requires them.

**Review by Hoikas — COMMENTED** [review]
This PR covers an odd set of topics. I think we would prefer to see changesets grouped by topic where possible. For example, one pull request for using `constexpr`, a separate pull request for converting iterator based loops to range-based loops, etc. This helps us to review changes and isolate regressions when applicable.

File: Sources/Plasma/Apps/plClient/win32/plClient_Win.cpp (hunk 1 above)
Comment (jfmherokiller) [reply Sources/Plasma/Apps/plClient/win32/plClient_Win.cpp:102]: i was thinking of adding the = but wasnt sure if it would change the code

**Pull request closed without merging**